### PR TITLE
Add loader for hugging face datasets.

### PR DESCRIPTION
**Pull request by Sid-Data-Universe**

### CODE REVIEW

File: finetune/datasets/hugging_face/hugging_face_subset_loader.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            for _ in range(self.batch_size):
+                batch.append(torch.tensor(self.buffer[: self.sequence_length]))
+                self.buffer = self.buffer[self.sequence_length :]
+            yield torch.stack(batch)
+
+    def __next__(self):
+        batch = []
+        for _ in range(self.batch_size):
+            batch.append(torch.tensor(self.buffer[: self.sequence_length]))
+            self.buffer = self.buffer[self.sequence_length :]
```
Comment (Sid-Data-Universe) [inline finetune/datasets/hugging_face/hugging_face_subset_loader.py:100]: Buffer could be changed to just a list of batches now given that each row is just one batch.

If doing so we could also avoid padding and just use truncation as well. Unsure if we will want to always have batches be exactly sequence length or not though in which case keeping the padding makes sense.

**Review by RusticLuftig — APPROVED** [review]
